cpp/src/hash/hashing.cu

```diff
@@ -99,7 +99,7 @@ std::unique_ptr<column> hash(table_view const& input,
       return serial_murmur_hash3_32<MurmurHash3_32>(input, seed, stream, mr);
     case (hash_id::HASH_SPARK_MURMUR3):
       return serial_murmur_hash3_32<SparkMurmurHash3_32>(input, seed, stream, mr);
-    default: return nullptr;
+    default: CUDF_FAIL("Unsupported hash function.");
```

Comment:
This change is to align this behavior with other functions that fail on unsupported enum values.